core/services/configuration_generator.rb

```diff
@@ -62,6 +62,10 @@ def self.generate_product_config(repos, product_name, product, box, repo)
     repo_file_name = repos.repo_file_name(product_name)
     config['repo_file_name'] = repo_file_name unless repo_file_name.nil?
     config['node_name'] = product['node_name'] unless product['node_name'].nil?
+    if repos.need_product_license?(product_name)
+      config['license'] = repos.product_license(product_name)
+      raise "License for product #{product['name']} not found" if config['license'].nil?
```

Comment:
If the handling of the Exception is not inside the ConfigurationGenerator, then this approach to solve errors is not appropriate.